### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by fulrose**
Our team have tried refactoring some code .

This request did not include test cases we tried earlier. 

If you want more detail, please feel free to comment.

### CODE REVIEW

**Comment by EmmanuelMess** [comment]
Could you pass a report on the changes?

**Comment by TranceLove** [comment]
I suppose you are a team of students trying to work on some code that runs in real life? Welcome aboard.

Can't say too much until the changes are even with our current `master`. Please resolve the conflicts first.

Additionally, some recommendations before actually looking into the changes:
 * squash the commits down to smaller numbers wherever possible - start with commits that do similar things. I had been bashed once when I submitted my bigger piece of work (see #863), so I think it'd be a good idea for you too. As long as your teammates have the full commit log or it's on a separate branch, it shouldn't hurt
 * give more details about the changes you made in the pull request. I suppose your changes wouldn't be able to summarise with a single "refactoring", but something like "code cleanups" with summary of changes will help everyone a lot

**Comment by EmmanuelMess** [comment]
Maybe it would be better if you encapsulated changes to a certain part of the app (TextManager, Filesystem, UI) in different PRs. Not saying you HAVE to, but if you can please do.

**Comment by TranceLove** [comment]
@fulrose A friendly reminder: you will want to resolve the conflicts with `master` first, or you and your teammates' effort will start to smell :)

**Comment by EmmanuelMess** [comment]
Tell us when you are ready for review

File: app/src/test/java/com/amaze/filemanager/filesystem/compressed/CompressedHelperTest.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        But .gz makes fault.
+     */
+    @Test
+    public void isFileExtractableFaultTest1() throws Exception{
+        assertTrue(CompressedHelper.isFileExtractable("/test/test.gz"));
+    }
+
+    /* I think this make fault.
+        If some user or programmer change file file extension and user downloads that file,
+        or some programmer make some unpopular extension such as .~~zip, .~~rar
```
Comment (EmmanuelMess) [inline app/src/test/java/com/amaze/filemanager/filesystem/compressed/CompressedHelperTest.java]: [Apache Tika](http://tika.apache.org/) will fix this when it is implemented.

**Comment by EmmanuelMess** [comment]
I'm closing this as it is not an improvement, although addition of parts of this PR can still be done.

**Pull request closed without merging**